src/ShopifyApp/Middleware/AuthShop.php

```diff
@@ -48,7 +48,7 @@ protected function validateShop(Request $request)
         $session = new ShopSession();
 
         // Grab the shop's myshopify domain from query or session
-        $shopDomainParam = $request->get('shop');
+        $shopDomainParam = $request->get('shop') ?? $request->header('shop');
```

Comment:
1. This would allow injection and account takeover via request header spoofing
2. `Shop:` is not a good header name, it would have to be something like X-Shopify-Shop or something (might be reserved)

Reply:
1. no because of the `$session->isValid()` in the same file

2. agree we can rename it

Reply:
3. This would mean all devs would have to make this change. This is not the right way to implement such a change because it means it's broken by default

Reply:
https://stackoverflow.com/questions/187655/are-https-headers-encrypted

Reply:
4. Blocks users from being able to simply POST a `form` element - this would mean everyone would have to use a library to post forms instead of just using forms how they should be

Reply:
> And by the way : https://stackoverflow.com/questions/187655/are-https-headers-encrypted

That doesn't have anything to do with anything though, it's not about the value being exposed it's about being able to arbitrarily send anything in that value

Reply:
4. in laravel $request->get() is also fetching post params, you can set the shop are are hidden field like you do with csrf-token from laravel

Reply:
> 3. This would mean all devs would have to make this change. This is not the right way to implement such a change because it means it's broken by default

thats correct in generell but the only think we do here is to also allow to pass the shop in the header instead only check for cookie/get/post param at all

Reply:
@StefanNeuser see https://github.com/ohmybrew/laravel-shopify/pull/298 for my fix